### PR TITLE
Implement S3 feed publisher

### LINKED ISSUE #3: Implement a published for S3
See the example in the Readme

### PR DESCRIPTION
**Pull request by nirebu**
Closes #3 

This PR aims implement a feed publisher for AWS S3. With this we'll be able to upload the generated feeds to S3 right after generation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,11 +77,36 @@ And then periodically run it to keep the data fresh
...
-- Rails cache
-- Static file
-- FTP
+#### S3
+
+Configuration, inside an initializer:
+
+```ruby
+  Aws.config[:profile] = 'my-profile'
+  s3 = SolidusFeeds::Publishers::S3.new(object_key: "foo/bar.txt", bucket: "my-bucket")
```
Comment (aldesantis) [inline README.md:86]: Why the weird indentation here?

File: README.md [hunk 2]
```diff
@@ -77,11 +77,36 @@ And then periodically run it to keep the data fresh
...
+```
+
+```ruby
+  Aws.config[:profile] = 'my-profile'
+  s3 = SolidusFeeds::Publishers::S3.new(object_key: "foo/bar.txt", bucket: "my-bucket", client: Aws::S3::Client.new(…)) # see docs
```
Comment (aldesantis) [inline README.md:91]: Same here.

File: README.md (hunk 1 above)
Comment (nirebu) [reply README.md]: Thanks for catching the typo

File: README.md (hunk 2 above)
Comment (nirebu) [reply README.md]: Same as above ☝️

**Review by aldesantis — APPROVED** [review]

**Pull request merged**